File: AcademicPlanBackend/src/main/java/com/bntu/fitr/poit/zholudev/diplom/entity/Standard.java
package com.bntu.fitr.poit.zholudev.diplom.entity;

import javax.persistence.*;
import javax.persistence.Entity;

@Entity
@Table(name = "standards")
public class Standard {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;

    public Standard() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: AcademicPlanFrontend/src/app/entity/standard.ts
export class Standard {
  id: number;
  name: string;

  static clone(standard: Standard): Standard {
    let cloneStandard: Standard = new Standard();
    cloneStandard.id = standard.id;
    cloneStandard.name = standard.name;
    return cloneStandard;
  }
}
